inference-engine/src/snippets/include/snippets/generator.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <transformations_visibility.hpp>
+#include "snippets_isa.hpp"
+
+namespace ngraph {
+namespace snippets {
+
+using code = const uint8_t *;
+using RegInfo = std::pair<std::vector<size_t>, std::vector<size_t>>;
+
+TRANSFORMATIONS_API auto getRegisters(std::shared_ptr<ngraph::Node>& n) -> ngraph::snippets::RegInfo;
+
+class Emitter {
+public:
+    Emitter(const std::shared_ptr<ngraph::Node>& n) {
+    }
+
+    virtual void emit_code(const std::vector<size_t>& in,
+                           const std::vector<size_t>& out,
+                           const std::vector<size_t>& pool = {},
+                           const std::vector<size_t>& gpr  = {}) const = 0;
+
+    virtual void emit_data() const {
+    }
+};
+
+class TargetMachine {
```

Comment:
I am not sure that we need to have internal classes (I think it is internal because we don't export it) in public API. Can we move these classes to the source folder?

Reply:
@ilyachur, those are not internal classes 😄  those a classes for plugin (backend in terms of code generation) to override to. TargetMachine and Emitter. It seems I need an export Marco for them 🤦